vertical-pod-autoscaler/pkg/utils/vpa/api.go

```diff
@@ -125,11 +127,45 @@ func stronger(a, b *vpa_types.VerticalPodAutoscaler) bool {
 }
 
 // GetControllingVPAForPod chooses the earliest created VPA from the input list that matches the given Pod.
-func GetControllingVPAForPod(pod *core.Pod, vpas []*VpaWithSelector) *VpaWithSelector {
+func GetControllingVPAForPod(pod *core.Pod, vpas []*VpaWithSelector, ctrlFetcher controllerfetcher.ControllerFetcher) *VpaWithSelector {
+
+	var ownerRefrence *meta.OwnerReference
+	for i := range pod.OwnerReferences {
+		r := pod.OwnerReferences[i]
+		if r.Controller != nil && *r.Controller {
+			ownerRefrence = &r
+		}
+	}
+	if ownerRefrence == nil {
+		// If the pod has no ownerReference, it cannot be under a VPA.
```

Comment:
Is that a behavior change?
If so I think we need API review and make sure we version the change accordingly.

With the prospect of in-place pod updates, @dbenque what do you think about this rule. I'm wondering if in the future it might make sense to have VPAs for single pods to still allow growing them?
Or would we expect people to still have them behind a Deployment with replica=1?

Reply:
> Is that a behavior change?

It changes the behaviour like any bug fix would do. Today a VPA can adopt orphan pods or even pods of another deployment if ever the labelSelectors overlap. This PR will prevent such "unexpected" behaviours.

> I'm wondering if in the future ...

When it comes to VPA (in-place update or not), a pod (or a set of pods) is expected to be behind a Deployment or any other controller (Statefulset, DaemonSet or exposing Scale subresource). This is a direct consequence of or VPA definition: [it uses targetRef](https://github.com/kubernetes/autoscaler/blob/7e95c7e63a5f0773e51bc484b1f7dbe802fe668a/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1/types.go#L76-L87)

A single pod has to be behind a Deployment with replica=1 or any other custom controller implementing scale if we want to be able to apply VPA recommendation to it.

Reply:
Re API review.

Not aware of any API review process for this sig. TBH, I believe many PRs change behaviours of VPA. @voelzmo and @mwielgus is there any API review process for this sig? Do you think a change like this PR requires a review?

Reply:
Actually this PR was already discussed in the sig meeting. @dbenque is there any remaining concerns from the sig meeting that needs to be addressed? If not it is good to me.